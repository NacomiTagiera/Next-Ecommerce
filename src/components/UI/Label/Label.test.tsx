import { render, screen } from "@testing-library/react";

import { Label } from ".";

describe("Label", () => {
	it("renders with default props", () => {
		const { asFragment } = render(<Label />);
		expect(asFragment()).toMatchSnapshot();
	});

	it("assigns class names by properties", () => {
		render(
			<Label color="error" visibility="shown">
				Label
			</Label>,
		);
		expect(screen.getByText("Label")).toHaveClass(
			"block text-sm font-medium leading-6 text-red-800",
		);
	});

	it("applies default class names when no props are provided", () => {
		render(<Label>Label</Label>);
		expect(screen.getByText("Label")).toHaveClass(
			"block text-sm font-medium leading-6 text-zinc-900",
		);
	});

	it("injects native label properties", () => {
		render(<Label data-testid="Test" />);
		expect(screen.getByTestId("Test")).toBeVisible();
	});
});
