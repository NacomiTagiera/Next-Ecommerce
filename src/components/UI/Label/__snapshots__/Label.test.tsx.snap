// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders with default props 1`] = `
<DocumentFragment>
  <label
    class="text-zinc-900 block text-sm font-medium leading-6"
  />
</DocumentFragment>
`;
